### DISCUSSION TITLE
Ability to combine referrer domains

### DISCUSSION THREAD

**Discussion by d19dotca** [umami-software/umami#1271] [General]
I'd love to somehow combine referrers in the metrics based on domains, such as counting google.ca; google.com; google.co.uk and other related Google domains can be a single line item in the Referrers list, is that possible to do at all? Ideally my client's care about how many referrals came from Google searches, and some of my client sites get many different Google domains coming in as referrers, was hoping to have a way to combine them all for simplicity sake so they don't have to count them all up manually. 

Any ideas if this is already possible, and if so then how I can do that? If not already possible, is it possible to add this as a feature request down the road?

**Comment by mikecao** [comment]
It's not possible now because those are all different domains technically. We could possibly use the middleware functions to allow you to customize the payload. It's not a very common use case so it would have to be custom code.

**Reply by d19dotca** [reply]
Interesting. I'm surprised that's considered not to be a very common use-case when the other analytics tools all seem to do it (Fathom, Plausible, Matomo, etc). It generally would make the most sense, as site owners generally don't care which specific Google domain the request came from, just that it came from Google and won't want to have to add them all up separately to get the total count.

**Comment by mikecao** [comment]
Can you show me where these other analytics do that? I'd like to see how they do it and see if we can implement it.

**Reply by d19dotca** [reply]
I'm not entirely sure where example code would be but if you look at the demo of Fathom or Matomo or even the Plausible demos... you'll see that the referrers/sources are grouped by type for the common ones like Google, Twitter, GitHub, and DuckDuckGo for example, which all remove the need to know specifically what domain of each one came from. And I'm pretty sure their reasoning is the same as mine behind this feature request... simplicity. Customers / end-users simply don't care (in the vast majority of cases anyways) how many came specifically from Google.com, Google.ca, Google.fr etc for example, as far as they're concerned in that example they just want the total of all Google domains combined. 

Some demo links are below: 

- https://demo.matomo.cloud/index.php?module=CoreHome&action=index&idSite=1&period=day&date=yesterday#?idSite=1&period=day&date=yesterday&category=General_Visitors&subcategory=Live_VisitorLog

- https://app.usefathom.com/share/deasaicp/hilarious+platypus#/?filters=%5B%5D&range=last_7_days&site=114903


- https://plausible.io/plausible.io

Hopefully that helps. Would definitely love to see that improved for end-users.

**Reply by rohandebsarkar** [reply]
I tried to search through the Plausible code to find how they group referrers, found this file https://github.com/plausible/analytics/blob/master/priv/ref_inspector/referers.yml, where all the domain data is stored. Apparently by reading the README in the same directory, it seems that they extracted this data from this repository https://github.com/snowplow-referer-parser/referer-parser.

Searching on the internet, I found this NPM package, https://www.npmjs.com/package/referer-parser, which extracts data from that repository and returns the marketing data (such as the Company/Organization, etc.) of the domain.
